### PR TITLE
Govcloud support

### PR DESCRIPTION
**Pull request by shivknight**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,39 +1,82 @@
+switch (env.JENKINS_ENV) {
```
Comment (NachoRaik) [inline Jenkinsfile]: Great add-on! It will save us from maintaining 2 versions of the pipeline :)

File: Jenkinsfile [hunk 2]
```diff
@@ -1,39 +1,82 @@
...
+        choice(name: 'Environment', choices: "\n" + supportedEnvs.join("\n"), description: 'Choose the environment where to publish the policy')
+
+        booleanParam(name: 'Snapshot', defaultValue: false, description: 'If selected, it will be published as snapshot. Otherwise, as a release')
 
-        booleanParam(name: 'Snapshot', defaultValue: false, description: 'Publish as release or as snapshot')
+        string(name: "ReleaseDescriptor", defaultValue: ' ', description: 'Relative path in ROM where the YAML descriptor is located')
+
+    }
 
+    environment {
```
Comment (NachoRaik) [inline Jenkinsfile]: Now that we have the switch env, we should put this there too

File: Jenkinsfile [hunk 3]
```diff
@@ -1,39 +1,82 @@
+switch (env.JENKINS_ENV) {
+  case "build-prod":
+    supportedEnvs = ['qax', 'stgx', 'prod']
+    managed_file_maven_settings = 'api-gateway-maven-settings'
+    github_creds_id = 'GH_TOKEN'
+    break
+  case "build-gov":
+    supportedEnvs = ['gstg', 'gprod']
+    managed_file_maven_settings = 'mulesoft-global-mvn-settings'
+    def github_creds_id = 'github'
```
Comment (NachoRaik) [inline Jenkinsfile]: For consistency,
```suggestion
   github_creds_id = 'github'
```

File: Jenkinsfile [hunk 4]
```diff
@@ -1,39 +1,82 @@
...
+                    sh "mkdir -p /root/.m2"
+                    sh "cp $MAVEN_SETTINGS /root/.m2/settings.xml"
+                }
+            }
+        }
 
-        stage('Publish') {
+        stage('Clone ROM') {
+            environment {
+                GITHUB_CREDS = credentials('mulesoft-gov-automation')
```
Comment (NachoRaik) [inline Jenkinsfile]: The `'mulesoft-gov-automation'` it's just for govCloud. Should also go to the switch then

**Comment by shivknight** [comment]
Please test on `jenkins-onprem`. If it works correctly there, this PR should be good to merge

**Pull request merged**